diff --git lib/internal/modules/esm/loader.js lib/internal/modules/esm/loader.js
index ffe82b37..01edb6d7 100644
--- lib/internal/modules/esm/loader.js
+++ lib/internal/modules/esm/loader.js
@@ -868,6 +868,11 @@ class ModuleLoader {
     }
     importMetaInitializer ??= require('internal/modules/esm/initialize_import_meta').initializeImportMeta;
     meta = importMetaInitializer(meta, context, this);
+    Object.defineProperty(meta, 'cache', {
+      configurable: false,
+      enumerable: true,
+      get: () => this.loadCache,
+    });
     return meta;
   }
 
